packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -6,6 +6,55 @@ import 'package:meta/meta.dart';
 
 import 'print.dart';
 
+/// The various priority levels used to filter which diagnostics are shown and
+/// omitted.
+///
+/// Trees of Flutter diagnostics can be very large so filtering the diagnostics
+/// shown matters. Typically filtering to only show diagnostics with at least
+/// level [debug] is appropriate.
+enum DiagnosticLevel {
+  /// Diagnostics that should not be shown.
+  ///
+  /// For example, property values that convey information which is completely
+  /// redundant with other properties or properties where the value is
+  /// misleading or hard to understand.
+  hidden,
+
+  /// Trivial or redundant diagnostics information.
+  ///
+  /// Use this level for diagnostic properties that match their default value
+  /// and other cases where showing a diagnostic would not add value.
+  fine,
```

Comment:
should we merge hidden and fine? their descriptions sound very similar.

Reply:
I think the problem was with the quality of my description of the three levels. I've written more to clarify the difference between hidden, fine, and debug.
TL:DR;
hidden: property display may be ugly or the value may simply be misleading (e.g. some other property takes priority so the property has no impact.
fine: property display is great but the value is known to be uninteresting.
debug: property value is interesting but it doesn't meet the bar to be a higher priority level likely due to being very verbose.